pyam/__init__.py

```diff
@@ -11,25 +11,26 @@
 from pyam.datareader import read_worldbank  # noqa: F401
 
 logger = logging.getLogger(__name__)
-logger.addHandler(logging.NullHandler())
 
-# in Jupyter notebooks: disable autoscroll and set logger to info
+# in Jupyter notebooks: disable autoscroll and set-up logging
 try:
-    get_ipython().run_cell_magic(u'javascript', u'',
-                                 u'IPython.OutputArea.prototype._should_scroll = function(lines) { return false; }')
-
-    logger.setLevel(logging.INFO)
-
-    stderr_info_handler = logging.StreamHandler()
-    formatter = logging.Formatter('%(name)s - %(levelname)s: %(message)s')
-    stderr_info_handler.setFormatter(formatter)
-    logger.addHandler(stderr_info_handler)
-
-    log_msg = (
-        "Running in a notebook, setting `{}` logging level to `logging.INFO` "
-        "and adding stderr handler".format(__name__)
-    )
-    logger.info(log_msg)
+    from ipykernel.zmqshell import ZMQInteractiveShell
+    from IPython import get_ipython
+
+    shell = get_ipython()
+    if isinstance(shell, ZMQInteractiveShell):
+        shell.run_cell_magic(
+            u'javascript', u'',
+            u'IPython.OutputArea.prototype._should_scroll = function(lines) { return false; }'
+        )
```

Comment:
The cell magic does not seem to have any effect for me! What is it's purpose? Should i remove it?

Reply:
By default, Jupyter notebook switches to scrolling within a cell output if it's too long (more than 100 lines).

Try running `[i for i in range(0, 110)]` in a completely fresh Jupyter notebook instance (not just restarted kernel of one notebook) and compare the output before and after running that cell magic.

@gidden had some very strong feelings about auto-scrolling in notebooks in the past (he strongly preferred switching it off), so I would not remove that section.

Reply:
Ok, not a problem in jupyterlab, but I see the issue in the notebook. And I definitely wouldn't want to incur @gidden's wrath.